### ISSUE TITLE
repo is not indexed

### ISSUE THREAD

**Issue by Seasawher** [leanprover/reservoir#26]
repo https://github.com/lean-ja/lean-by-example has not been indexed by Reservoir

I think this repo satisfies the inclusion criteria https://reservoir.lean-lang.org/inclusion-criteria.

**Comment by tydeu** [comment]
@Seasawher I have now figured out the problem. Lean by example is a repository generated from a template. GitHub treats categorizes thus repositories as forks from the perspective of repository / code search. Reservoir does not include forks in its indexing (and cannot, as this would produce to many results), thus it does not pick up Lean by example.

It is probably worth clarify this in the inclusion criteria.

**Comment by Seasawher** [comment]
oh, thanks

**Comment by Seasawher** [comment]
Thank you. In the future, will there be a feature that allows a repository to be indexed by Reservoir even if it is a fork?

**Comment by tydeu** [comment]
@Seasawher Yes, probably. It will likely be part of a manual package registration process that will also solve #13.

**Comment by Seasawher** [comment]
Thanks!

**Comment by Seasawher** [comment]
@tydeu 

why this repo is included? https://reservoir.lean-lang.org/@Joonas-vonlerber/Game

**Comment by tydeu** [comment]
@Seasawher I am not sure. 😕 When search repositories the way Riposte does manually, it does not appear in the search, but it apparently appeared [once](https://github.com/leanprover/reservoir/actions/runs/8687665342/job/23821620500#step:5:615) and got indexed then. I am not sure what caused to appear in the search at the time and not now. As a generated repository, it theoretically should not have appeared in the search according to GitHub's apparent rules. Maybe its owner created it as a normal repository first and then recreated as a generated one and that is why it got indexed? Alternatively maybe there is some race condition in GitHub were the search saw it as normal repository before it got recorded as fork? I have no clue.

**Comment by Seasawher** [comment]
Could the following repositories be added to Reservoir?

* https://github.com/lean-ja/lean-by-example
* https://github.com/lean-ja/lean99

**Comment by tydeu** [comment]
@Seasawher Sorry for the late reply! I would be happy to do this; however, I do not currently have a process for manually adding repositories to the Reservoir index, so this will have until I have the time to do so.

**Comment by tydeu** [comment]
@Seasawher Both of your packages were indexed back in late 2024. I keep this issue open as a reminder to implement manual package registration (as it was one of the first requests to index a package). An API for this is now implemented (#96) 🎉 , so I am closing this issue. Thanks for being patient and raising the initial awareness of this issue! ❤️
